### PR TITLE
Updated user guide, morphed a bit of GUI, created basic student classes, modified tests

### PR DESCRIPTION
**Pull request by m-aslam-mj2**
- updated user guide and added student management specific features
- morphed a bit of the GUI to reflect wireframe design
- started writing classes for student management
- removed or modified several tests pending revised test cases for new changes.

P.S. really gotta spread out the PRs.

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -250,6 +252,7 @@ Assigns the student with matric no. `A0171234B` as a TA.
 * `module assign e/e0191234@u.nus.edu` +
 Assigns the student with email `e0191234@u.nus.edu` as a TA.
 
+=======
```
Comment (harriuscai) [inline docs/UserGuide.adoc]: This particular line seems to have cut off the subsequent sections from the table of contents

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -50,83 +50,85 @@ image::Ui.png[width="790"]
...
 
 ****
-* Edits the person at the specified `INDEX`. The index refers to the index number shown in the displayed person list. The index *must be a positive integer* 1, 2, 3, ...
-* At least one of the optional fields must be provided.
-* Existing values will be updated to the input values.
-* When editing tags, the existing tags of the person will be removed i.e adding of tags is not cumulative.
-* You can remove all the person's tags by typing `t/` without specifying any tags after it.
+* Edits the person at the specified INDEX. The index refers to the index number shown in the displayed student list. The index must be a must be a positive integer 1,2,3,...
+* At least one of the optional fields must be provided
+* Existing values will be updated to the input values
```
Comment (harriuscai) [inline docs/UserGuide.adoc]: The points of interest should be above the examples

File: src/main/java/seedu/address/model/student/Student.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+
+
+
+/**
+ * Represents a student in the LMS.
```
Comment (harriuscai) [inline src/main/java/seedu/address/model/student/Student.java]: Shall we start referring to our system by its name, Trajectory?

File: src/main/java/seedu/address/model/student/Student.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+     * @param email
+     * @param address
+     * @param tags
+     */
+
+    private String courseCode;
+    private String matricNo;
+
+    public Student(Name name, Phone phone, Email email, Address address,
+                   Set<Tag> tags, String courseCode, String matricNo) {
```
Comment (harriuscai) [inline src/main/java/seedu/address/model/student/Student.java]: We will need to verify with Steph if this indentation is acceptable. When I asked Akshay about it, he told me to follow the 8 spacing rule for word-wrapping strictly as the code style is graded by a bot.

**Comment by m-aslam-mj2** [comment]
[User Guide: Move points of interest to above examples (refer to the other features in the guide for the formatting)](https://trello.com/c/hlAheQLT/52-user-guide-move-points-of-interest-to-above-examples-refer-to-the-other-features-in-the-guide-for-the-formatting)

**Pull request merged**